sdk-setup/src/mb2

```diff
@@ -1856,104 +1856,107 @@ run_build_requires__process_args() {
+    if inside_build_engine; then
+        local shared_key=${OPT_SHARED_DIR}/gnupg/${OPT_PACKAGE_SIGN_USER}.key
+        if [[ ! -f $shared_key ]]; then
+            fatal "Cannot sign packages: Internal error: No GPG key shared for the user '$OPT_PACKAGE_SIGN_USER'."
+        fi
 
-    if [[ $OPT_PACKAGE_SIGN_USER ]]; then
+        local shared_key_id=$(gpg2 --with-colons --list-packets "$shared_key" \
+                |sed -n 's/:user ID packet: "\(.*\)"/\1/p')
+        [[ $shared_key_id ]] || return
 
-        if inside_build_engine; then
-            local shared_key=${OPT_SHARED_DIR}/gnupg/${OPT_PACKAGE_SIGN_USER}.key
-            if [[ ! -f $shared_key ]]; then
-                notice "Cannot sign packages: Internal error: No GPG key shared for the user '$OPT_PACKAGE_SIGN_USER'."
-                return 1
+        local key_fingerprints=$(gpg2 --with-colons --list-secret-keys --fingerprint "$shared_key_id" \
+                |awk -F: '$1 == "fpr" { print $10 }')
+        local key_fingerprint=
+        for key_fingerprint in $key_fingerprints; do
+            if ! gpg2 --batch --delete-secret-and-public-key "$key_fingerprint"; then
+                fatal "Cannot sign packages: Internal error: Failed to delete old GPG key copy '$key_fingerprint'."
             fi
+        done
 
-            local shared_key_id=$(gpg2 --with-colons --list-packets "$shared_key" \
-                    |sed -n 's/:user ID packet: "\(.*\)"/\1/p')
-            [[ $shared_key_id ]] || return
-
-            local key_fingerprints=$(gpg2 --with-colons --list-secret-keys --fingerprint "$shared_key_id" \
-                    |awk -F: '$1 == "fpr" { print $10 }')
-            local key_fingerprint=
-            for key_fingerprint in $key_fingerprints; do
-                if ! gpg2 --batch --delete-secret-and-public-key "$key_fingerprint"; then
-                    notice "Cannot sign packages: Internal error: Failed to delete old GPG key copy '$key_fingerprint'."
-                    return 1
-                fi
-            done
-
-            if ! gpg2 --quiet --import "${shared_key}"; then 
-                notice "Cannot sign packages: Internal error: Failed to import GPG key from file '$shared_key'."
-                return 1
-            fi
+        if ! gpg2 --quiet --import "${shared_key}"; then 
+            fatal "Cannot sign packages: Internal error: Failed to import GPG key from file '$shared_key'."
         fi
+    fi
 
-        if ! gpg2 --list-secret-keys "$OPT_PACKAGE_SIGN_USER" &>/dev/null; then
-            notice "Cannot sign packages: No GPG key found for the user '$OPT_PACKAGE_SIGN_USER'."
-            return 1
-        fi
+    if ! gpg2 --list-secret-keys "$OPT_PACKAGE_SIGN_USER" &>/dev/null; then
+        fatal "Cannot sign packages: No GPG key found for the user '$OPT_PACKAGE_SIGN_USER'."
+    fi
 
-        extra_args+=(--define "_gpg_name $OPT_PACKAGE_SIGN_USER")
+    SIGN_EXTRA_ARGS+=(--define "_gpg_name $OPT_PACKAGE_SIGN_USER")
```

Comment:
I would move this as a local variable to `sign_rpms_native` so that one can spot it all with one glance.